src/main/java/net/neoforged/neoforge/common/extensions/IBlockExtension.java

```diff
@@ -112,6 +112,40 @@ default float getFriction(BlockState state, LevelReader level, BlockPos pos, @Nu
         return self().getFriction();
     }
 
+    /**
+     * Gets the speed factor at the given location at the given state. Normally
+     * between 0 and 1.
```

Comment:
I would emphasize more that the speed factor is multiplied to the XZ components of the current movement vector. This is ignored if the player is flying. Additionally, for the typical entity, the block speed factor is lerped between the factor and 1 based on the entity's movement efficiency (which is by default 0). This means that speed factors greater than 1 will decrease when movement efficiency is increased. Currently, this only affects the soul speed enchantment, but as it is an attribute, it should be mentioned.